scio-test/src/main/scala/com/spotify/scio/testing/BigTableMatchers.scala

```diff
@@ -0,0 +1,126 @@
+
+import com.google.bigtable.v2.Mutation
+import com.google.bigtable.v2.Mutation.MutationCase
+import com.google.protobuf.ByteString
+import com.spotify.scio.values.SCollection
+import org.scalatest.matchers.{MatchResult, Matcher}
+
+import scala.reflect.ClassTag
+
+/**
+  * Trait with ScalaTest [[org.scalatest.matchers.Matcher Matcher]]s for
+  * [[com.spotify.scio.values.SCollection SCollection]]s specific to BigTable output.
+  */
+trait BigTableMatchers extends SCollectionMatchers {
+  type BTRow = (ByteString, Iterable[Mutation])
+  type BTCollection = SCollection[BTRow]
+
+  // Provide an implicit BT serializer for common cell value type String
+  implicit val stringBTSerializer: String => ByteString = ByteString.copyFromUtf8
+
+  /** Check that the BT collection contains only the given keys, in any order. */
+  def containRowKeys(expectedKeys: String*): Matcher[BTCollection] =
+    new Matcher[BTCollection] {
+      override def apply(left: BigTableMatchers.this.BTCollection): MatchResult = {
+        containInAnyOrder(expectedKeys).apply(left.keys.map(_.toStringUtf8))
+      }
+    }
+
+  /** Check that the BT collection contains only the given column families, unique, in any order. */
+  def containColumnFamilies(expectedCFs: String*): Matcher[BTCollection] =
+    new Matcher[BTCollection] {
+      override def apply(left: BigTableMatchers.this.BTCollection): MatchResult = {
+        val foundCFs = left.flatMap {
+          case (key, cells) =>
+            cells.map(_.getSetCell.getFamilyName)
+        }
+
+        containInAnyOrder(expectedCFs).apply(foundCFs.distinct)
+      }
+    }
+
+  /**
+    * Check that the BT collection contains a cell with the given row key, column family, and
+    * deserialized cell value. Column qualifier defaults to the same as column family.
+    */
+  def containSetCellValue[V: ClassTag](key: String, cf: String, value: V)
+                                    (implicit ser: V => ByteString): Matcher[BTCollection] =
+    containSetCellValue(key, cf, cf, value)
+
+  /**
+    * Check that the BT collection contains a cell with the given row key, column family,
+    * column qualifier, and deserialized cell value.
+    * @param key Row key the cell should be in
+    * @param cf Column family the cell should have
+    * @param cq Column qualifier the cell should have
+    * @param value Deserialized value of the set cell
+    * @param ser Serializer to convert value type V to ByteString for BT format
+    * @tparam V Class of expected value
+    * @return Whether the collection contains this cell, with no assumptions made about the
+    *         contents of the collection.
+    */
+  def containSetCellValue[V: ClassTag](key: String, cf: String, cq: String, value: V)
+                                    (implicit ser: V => ByteString): Matcher[BTCollection] =
+    new Matcher[BTCollection] {
+      override def apply(left: BTCollection): MatchResult = {
+        val flattenedRows = left.flatMap {
```

Comment:
Minor: `.flatMap { case ...`
Also `rowValue { cell => ... }` is fewer `{}` and probably cleaner.